packages/vsce/src/commands/showLogsCommand.ts

```diff
@@ -0,0 +1,122 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { commands, TreeView, window } from "vscode";
+import { CICSRegionTree } from "../trees/CICSRegionTree";
+import { findSelectedNodes } from "../utils/commandUtils";
+import { ProfileManagement } from "../utils/profileManagement";
+import { Gui } from "@zowe/zowe-explorer-api";
+import { getResource, ICMCIApiResponse } from "@zowe/cics-for-zowe-sdk";
+import * as vscode from "vscode";
+import { IProfileLoaded } from "@zowe/imperative";
+
+export async function findRelatedZosProfiles(cicsProfile: IProfileLoaded, zosProfiles: IProfileLoaded[]): Promise<IProfileLoaded> {
+  const baseForCicsProfile = await ProfileManagement.getProfilesCache().fetchBaseProfile(cicsProfile.name);
+
+  // sort profiles with zosmf ones first to make zosmf the default
+  // also filter so we only automatically pick z/os connections that have credentials associated
+  zosProfiles = zosProfiles.sort((prof) => (prof.profile.type === "zosmf" ? -1 : 1)).filter((prof) => prof.profile.user);
+
+  // filter out profiles that are not in the same base as the cics profile
+  const matchingProfiles: IProfileLoaded[] = [];
+  if (baseForCicsProfile) {
+    for (const profile of zosProfiles) {
+      if (baseForCicsProfile && baseForCicsProfile?.name === (await ProfileManagement.getProfilesCache().fetchBaseProfile(profile.name))?.name) {
```

Comment:
Asserted it exists earlier in the line

```suggestion
      if (baseForCicsProfile && baseForCicsProfile.name === (await ProfileManagement.getProfilesCache().fetchBaseProfile(profile.name))?.name) {
```